tests/src/test_cases/test_display.c

```diff
@@ -0,0 +1,130 @@
+#include "unity/unity.h"
+
+void setUp(void)
+{
+    /* Function run before every test */
+}
+
+void tearDown(void)
+{
+    /* Function run after every test */
+}
+
+struct display_area_test_set {
+    /* Parameters for setting up the display */
+    uint32_t width;
+    uint32_t height;
+    lv_color_format_t color_format;
+    lv_display_render_mode_t render_mode;
+    /* Parameters for testing */
+    uint32_t invalidated_width;
+    uint32_t invalidated_height;
+    uint32_t expected_buf0_size;
+};
+
+void test_get_drawbuf_size_double_buffered(void)
+{
+    static  LV_ATTRIBUTE_MEM_ALIGN uint8_t buf0[200 + LV_DRAW_BUF_ALIGN];
+    static  LV_ATTRIBUTE_MEM_ALIGN uint8_t buf1[200 + LV_DRAW_BUF_ALIGN];
+
+    lv_display_t * disp = lv_display_create(10, 20);
+    lv_display_set_color_format(disp, LV_COLOR_FORMAT_RGB888);
+
+    lv_display_set_buffers(disp, lv_draw_buf_align(buf0, LV_COLOR_FORMAT_RGB888), lv_draw_buf_align(buf1,
+                                                                                                    LV_COLOR_FORMAT_RGB888), 200, LV_DISPLAY_RENDER_MODE_PARTIAL);
+
+    TEST_ASSERT_EQUAL(200, lv_display_get_draw_buf_size(disp));
+}
+
+void test_get_drawbuf_size_single_buffered(void)
+{
+    static  LV_ATTRIBUTE_MEM_ALIGN uint8_t buf0[200 + LV_DRAW_BUF_ALIGN];
+
+    lv_display_t * disp = lv_display_create(10, 20);
+    lv_display_set_color_format(disp, LV_COLOR_FORMAT_RGB888);
+
+    lv_display_set_buffers(disp, lv_draw_buf_align(buf0, LV_COLOR_FORMAT_RGB888), NULL, 200,
+                           LV_DISPLAY_RENDER_MODE_PARTIAL);
+
+    TEST_ASSERT_EQUAL(200,  lv_display_get_draw_buf_size(disp));
+}
+
+static void exec_invalidated_drawbuf_size_test(const struct display_area_test_set * test_set)
+{
+    uint32_t buffer_size = lv_draw_buf_width_to_stride(test_set->width, test_set->color_format) * test_set->height;
+
+    uint8_t * buf0 = lv_malloc(buffer_size + LV_DRAW_BUF_ALIGN);
+
+    lv_display_t * disp = lv_display_create(test_set->width, test_set->height);
+    lv_display_set_color_format(disp, test_set->color_format);
+
+    lv_display_set_buffers(disp, lv_draw_buf_align(buf0, test_set->color_format), NULL, buffer_size,
+                           test_set->render_mode);
+    uint32_t invalidated_size = lv_display_get_invalidated_draw_buf_size(disp, test_set->invalidated_width,
+                                                                         test_set->invalidated_height);
+
+    TEST_ASSERT_EQUAL(test_set->expected_buf0_size, invalidated_size);
+    lv_free(buf0);
+    lv_display_delete(disp);
+}
+
+void test_get_invalidated_drawbuf_size_rgb888_partial()
+{
+    struct display_area_test_set test_set = {
+        .width = 10,
+        .height = 20,
+        .color_format = LV_COLOR_FORMAT_RGB888,
+        .render_mode = LV_DISPLAY_RENDER_MODE_PARTIAL,
+        .invalidated_width = 5,
+        .invalidated_height = 5,
+        .expected_buf0_size = LV_ROUND_UP(5 * 3, LV_DRAW_BUF_STRIDE_ALIGN) * 5,
```

Comment:
> Seems to me like in `display.c` the `LV_DRAW_BUF_ALIGN` is 1 for 64-bit test but in the `test_display.c` its `64` for some weird reason. Not sure what is going on there kisvegabor

Can you make the expected result be based on the configured `LV_DRAW_BUF_ALIGN`?

Reply:
Hmm not sure, I thought the `LV_DRAW_BUF_ALIGN` is only important for allocating the draw buffer correctly. It should not matter in regards to the invalidation size - right?

Reply:
I found the problem. The test fails with the VGLite test where `width_to_stride` is [overwritten](https://github.com/lvgl/lvgl/blob/master/src/draw/vg_lite/lv_vg_lite_utils.c#L654-L661) and doesn't use the `LV_DRAW_BUF_STRIDE_ALIGN`.

I suggest 

```suggestion
        .expected_buf0_size = lv_draw_buf_width_to_stride(5, test_set.color_format),
```

Reply:
When originially implementing this I looked at the impl of `lv_draw_buf_width_to_stride` and wondered why this was architectured in a way the the stride calculation functions are overwritable. Guess I now have my answer :D

Reply:
Yeah, GPUs can have specific stride requirements.